tests/platforms/windows/app/create/test_create.py

```diff
@@ -28,24 +28,39 @@ def test_unsupported_host_os(create_command, host_os):
 
 
 @pytest.mark.parametrize("host_arch", ["i686", "ARM64", "wonky"])
-def test_unsupported_arch(create_command, host_arch):
+def test_unsupported_arch(create_command, host_arch, first_app_config, capsys):
     """Windows commands can only run on x86-64."""
     create_command.tools.host_os = "Windows"
     create_command.tools.host_arch = host_arch
+    create_command.apps["first"] = first_app_config
 
     with pytest.raises(
         UnsupportedHostError,
         match=f"Windows applications cannot be built on an {host_arch} machine.",
     ):
-        create_command()
+        create_command.verify_host()
+
+    assert capsys.readouterr() == ("", "")
+
+    # External apps can be built on a different architecture, with a warning.
+    first_app_config.external_package_path = "external"
+    create_command.verify_host()
+    stdout, stderr = capsys.readouterr()
+    assert "WARNING: Possible architecture mismatch" in stdout
+    assert stderr == ""
+
+    # The warning is not repeated on cloned commands.
+    create_command.is_clone = True
+    create_command.verify_host()
+    assert capsys.readouterr() == ("", "")
```

Comment:
This feels like it should be a second test, rather than piggybacking on test_unsupported_arch